terraform/modules/cluster/main.tf

```diff
@@ -12,23 +12,31 @@ resource "openstack_compute_keypair_v2" "keypair" {
 }
 
 resource "openstack_containerinfra_cluster_v1" "cluster" {
-  name                 = var.cluster_name
-  cluster_template_id  = data.openstack_containerinfra_clustertemplate_v1.clustertemplate.id
+  name                = var.cluster_name
+  cluster_template_id = data.openstack_containerinfra_clustertemplate_v1.clustertemplate.id
 
-  master_count         = var.master_count
-  node_count           = var.node_count
+  master_count = var.master_count
+  node_count   = var.node_count
 
-  keypair              = openstack_compute_keypair_v2.keypair.id
-  flavor               = var.flavor_name
-  master_flavor        = var.master_flavor_name
+  keypair       = openstack_compute_keypair_v2.keypair.id
+  flavor        = var.flavor_name
+  master_flavor = var.master_flavor_name
 
-  #TODO: fix how we can specify labels?
-  #labels = {
-  #  min_node_count                      = var.node_count
-  #  max_node_count                      = var.max_node_count
-  #}
+  labels = merge(data.openstack_containerinfra_clustertemplate_v1.clustertemplate.labels,
+    {
+      min_node_count = var.node_count
+      max_node_count = var.max_node_count
+  })
+}
+
+resource "null_resource" "kubeconfig" {
+  triggers = {
+    kubeconfig = var.cluster_name
+  }
 
   provisioner "local-exec" {
-    command = "mkdir -p ~/.kube/${var.cluster_name}; openstack coe cluster config ${var.cluster_name} --dir ~/.kube/${var.cluster_name} --force"
+    command = "mkdir -p ~/.kube/${var.cluster_name}; openstack coe cluster config ${var.cluster_name} --dir ~/.kube/${var.cluster_name} --force; ln -sf ~/.kube/${var.cluster_name}/config ~/.kube/config"
```

Comment:
are you sure we should add that link to override the default in all directories?

Reply:
AH i find it useful but I guess other users might find it annoying, I will get rid of it.

Reply:
Done.